### PR TITLE
User games list

### PR DESCRIPTION
**Pull request by taphill**
### Description

Allows the FE to make a GET request on api/v1 /users/games which will return all of the games a user has added to their shelf.

### Closes issue(s)

#39 

### Testing

Tested both happy and sad paths

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [x] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [ ] I ran rubocop and fixed issues
- [x] I ran full test suite - all tests passing

### CODE REVIEW

File: app/controllers/api/v1/users/games_controller.rb [hunk 1]
```diff
@@ -4,6 +4,12 @@ module Api
   module V1
     module Users
       class GamesController < ApplicationController
+        def index
+          user_games = User.find(params[:user_id]).games
+
```
Comment (katemorris) [inline app/controllers/api/v1/users/games_controller.rb]: Might run rubocop, that'll fix this error, well the autofix on rbocop will

File: config/routes.rb [hunk 2]
```diff
@@ -3,8 +3,9 @@
 
   namespace :api do
     namespace :v1 do
-      delete 'users/games', to: 'users/games#destroy'
```
Comment (katemorris) [inline config/routes.rb:6]: What would happen if we changed all of these to be in the resources users? Not sure if you've tried that or not. Not a big deal either way.

File: db/migrate/20210112022620_change_age_range_to_string.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class ChangeAgeRangeToString < ActiveRecord::Migration[5.2]
+  def change
+    change_column :games, :age_range, :string
```
Comment (katemorris) [inline db/migrate/20210112022620_change_age_range_to_string.rb:3]: Thanks for doing this!

File: spec/requests/api/v1/users/games/request_spec.rb [hunk 4]
```diff
@@ -3,6 +3,85 @@
...
+        json_body = JSON.parse(response.body, symbolize_names: true)
+
+        expect(response.status).to eq(200)
+        expect(json_body).to be_a(Hash)
+        expect(json_body).to have_key(:data)
+        expect(json_body[:data]).to be_a(Array)
+        expect(json_body[:data].count).to eq(3)
+        expect(json_body[:data][0][:attributes]).to be_a(Hash)
+
+        json_body[:data].each do |item|
```
Comment (katemorris) [inline spec/requests/api/v1/users/games/request_spec.rb:28]: We can put all of this in the response checkers doc. Doesn't have to be now though. We can do it during test cleanup.

File: spec/requests/api/v1/users/games/request_spec.rb [hunk 5]
```diff
@@ -3,6 +3,85 @@
...
+          expect(item[:attributes]).to have_key(:num_players)
+          expect(item[:attributes][:num_players]).to be_a(String)
+
+          expect(item[:attributes]).to have_key(:age_range)
+          expect(item[:attributes][:age_range]).to be_a(String)
+        end
+      end
+    end
+
+    context 'when passed a user_id of a user with no games' do
```
Comment (katemorris) [inline spec/requests/api/v1/users/games/request_spec.rb:53]: Nice sad paths!

**Review by katemorris — CHANGES_REQUESTED** [review]
Just run rubo cop and check out my comments. Once rubocop is run, I am happy to merge this.

**Review by katemorris — APPROVED** [review]

File: config/routes.rb (hunk 2 above)
Comment (taphill) [reply config/routes.rb:6]: I have tried that but it turns the route into `/api/v1/users/:user_id/games` and not `/api/v1/users/games`

Last time I looked at the FE they were using the `/api/v1/users/games` route unless that changed?

Comment (katemorris) [reply config/routes.rb:6]: I told them that we were asking for the ID in the path for these as of this morning. It's Restful I think so I kept it that way.

**Review by katemorris — APPROVED** [review]
Looks great!

**Pull request merged**